R/make.R

```diff
@@ -1,4 +1,290 @@
 
+#' Create an igraph graph from a list of edges, or a notable graph
+#'
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' `graph()` was renamed to `make_graph()` to create a more
+#' consistent API.
+#' @inheritParams make_graph
+#' @keywords internal
+#' @export
+graph <- function(edges, ..., n = max(edges), isolates = NULL, directed = TRUE, dir = directed, simplify = TRUE) { # nocov start
+  lifecycle::deprecate_soft("2.0.0", "graph()", "make_graph()")
+  if (!missing(dir) && missing(directed)) {
+   directed <- dir
+  }
+
+  if (missing(simplify)) {
+    make_graph(edges = edges, n = n, isolates = isolates, directed = directed, ...)
```

Comment:
Would this fail if `edges` is a formula?

I don't like the interface of `make_graph()`, but deprecating it is an uphill battle. Can we instead export all relevant functions (e.g., `make_famous_graph()`), and supersede it?

Reply:
let's open another issue regarding make_graph()'s superseding, and I'll do my best to fix the current graph() here in this PR.